R/model-module-helpers.R

```diff
@@ -178,23 +178,17 @@ add_pmfs <- function(pmfs) {
   if (!is.list(pmfs)) {
     return(pmfs)
   }
-  lpmfs <- purrr::map_dbl(pmfs, length)
-  l <- sum(lpmfs)
-  conv <- rep(0, l)
-  conv[1:lpmfs[1]] <- pmfs[[1]]
-  for (s in 2:d) {
-    # P(Z = z) = sum_over_x(P(X = x) * P(Y = z - x)) # nolint
-    proc <- rep(0, l)
-    for (i in 1:l) {
-      for (j in 1:lpmfs[s]) {
-        if (i >= j) {
-          proc[i] <- proc[i] + pmfs[[s]][j] * conv[i - j + 1]
-        }
+  return(
+    Reduce(x = pmfs, f = function(conv, pmf) {
+      lc <- length(conv)
+      wd <- seq_len(lc) - 1
+      proc <- numeric(lc + length(pmf))
```

Comment:
as this drops the comment equation I think additional documentation may be needed. I can add this before merging or happy to see it added.

Reply:
Would you prefer the commented equation re-added, a plain text description, or would you prefer to add a more concise equation? The last I'd leave to you to be sure it's correct.

Reply:
Commented equation re-added in https://github.com/epinowcast/epinowcast/pull/187/commits/bdf172ebeff5eb4b3882cdab6f0e4a3d8abe228f.

Reply:
yup looks good. I think we could flesh out the documentation but happy to leave that for a future PR etc just having the commented equation back in is enough I think to hand hold users reading the code.